bmap/src/discarder.rs

```diff
@@ -1,6 +1,10 @@
-use crate::SeekForward;
-use std::io::Read;
-use std::io::Result as IOResult;
+use crate::{AsyncSeekForward, SeekForward};
+use async_trait::async_trait;
+use futures::executor::block_on;
+use futures::io::{AsyncRead, AsyncReadExt};
```

Comment:
Are we particularly against just depending on tokio in the bmap crate? Then you could just use tokio's AsyncRead & avoid the need for tokio-util entirely.

Reply:
We'd like to keep bmap crate as low level as possible, for it to have the less requirements in order to make it more compatible/exportable.

Reply:
ideally we'd like to keep the bmap create to use std only; it may be used in other platforms later, e.g. for #19 we'd like to be able to use the crate in an EFI environment.